### ISSUE TITLE
Integration of Find Multiple Hotspots not possible

### ISSUE THREAD

**Issue by petzi53** [h5p/h5p-interactive-book#43]
I just found out that I cannot integrate the content type "Find Multiple Hotspots" into Interactive Book. Is there any reason for this exclusion?

**Comment by otacke** [comment]
@petzi53 Yes. The content type was [created by Luke Muller](https://h5p.org/find-multiple-hotspots), not by the H5P core team. Unfortunately, he did neither implement the so called "[Question Type Contract](https://h5p.org/documentation/developers/contracts)" nor take care of initializing some variables (among other shortcomings). Especially the former is the reason why it cannot be included in compound content types. Please note that [Luke dropped further development](https://github.com/lukemuller/h5p-image-multiple-hotspot-question/blob/master/README.md). The H5P core team seems to have [adopted the content type](https://github.com/h5p/h5p-image-multiple-hotspot-question/commits/master), and there's also a [pull request to fix the things mentioned above](https://github.com/h5p/h5p-image-multiple-hotspot-question/pull/2) (you see there's a lot of stuff added), but I hope you understand that the H5P core team cannot take over maintenance for every content type that other authors abandon.

**Comment by petzi53** [comment]
@otacke Thank you for this detailed and helpful answer. Is there a similar reason why Advanced Fill the Blanks is not integrated into other content types? It is also excluded from Interactive Book as I have just noticed.

I am looking forward to your tutorial on how to integrate other content types hoping that I will learn how to make these kinds of adaptions by myself.

**Comment by otacke** [comment]
@petzi53 No, Advanced Fill in the Blanks seems to implement the Question Type Contract and seems to sanitize parameters. Must be a different reason.

**Comment by petzi53** [comment]
Sorry, still another question: Guess the Answer does not seem to display its header in the table of content of Interactive Book on the left side. So you can't jump to it when you a "Guess the Answer"-exercise as a subpage in a chapter. The only way to link to it with a specific URL is to make each "Guess the Answer"-exercise as its own chapter.

Another issue is: Guess the Answer is an interactive exercise because you can press the button "Click to see the answer".  But it comports like a page without tasks, e.g. progress is shown whenever the page is just called but without doing the exercise.
